src/balance/priority.go

```diff
@@ -0,0 +1,56 @@
+/**
+ * weight.go - weight balance impl
+ *
+ * @author Yaroslav Pogrebnyak <yyyaroslav@gmail.com>
+ */
+
+package balance
+
+import (
+	"../core"
+	"errors"
+	"math/rand"
+)
+
+/**
+ * Weight balancer
+ */
+type PriorityBalancer struct{}
+
+/**
+ * Elect backend based on weight strategy
+ */
+func (b *PriorityBalancer) Elect(context core.Context, backends []*core.Backend) (*core.Backend, error) {
+
+	if len(backends) == 0 {
+		return nil, errors.New("Can't elect backend, Backends empty")
+	}
+
+	/* count the backends with the same priority */
+	matchingPriority := 0
+	bestPriority := 0
+	for _, backend := range backends {
+		if (matchingPriority == 0 || backend.Priority < bestPriority) {
+			bestPriority = backend.Priority
+			matchingPriority = 1
+		} else if backend.Priority == bestPriority  {
+			bestPriority = backend.Priority
+			matchingPriority ++;
+		}
+
+	}
+
```

Comment:
Maybe, it would be better to calculate matching backend to return in the first loop, without looping the same backends again?

Reply:
arg. I lost my original comment. It essentially was saying that I think you need to look at every single backend because you dont know which has the lowest priority, and you dont know how many of those backends have the same priority.
(ie priority=10 could be the lowest, and there could be three servers with priority=10)
One way you could do it would be to build another array of "selectable_backends" as you did the first loop.
using the "priority=10, three backend servers" example - once you finish the first loop, the array would only contain the three server backends..  you know the size of the array and could randomly select from them.
so, still two loops required, and this solution would mean building new arrays and dumping them when a lower priority is found. i'm not sure how much more efficient it would be?

Reply:
Well, that makes sense. Ok, so the main question here is - what is the use case for distinct priority balancing method? We have a strong belief that priority should be implemented in coordination with weight, and preliminary support is already implemented -- gb fetches priority parameters. It would be nice if you implement your algorithm for that.

Reply:
I did think about that actually. its literally cut'n'pasting about 5 lines from the weight balancer.
I was trying to keep things simple..